liquidprompt

```diff
@@ -1802,6 +1825,9 @@ _lp_set_prompt()
         # end of the prompt line: double spaces
         LP_MARK="$(_lp_sr "$(_lp_smart_mark $LP_VCS_TYPE)")"
 
+        _lp_shorten_path   # set LP_PWD
+        _lp_shorten_vcs_path
```

Comment:
You are calling both `lp_shorten_path` and `_lp_shorten_vcs_path`. This is inefficient as only the result of one of them will matter.

1. Instead of calling both, `_lp_shorten_vcs_path` should delegate to `_lp_shorten_path` in case it does not fill `LP_PWD`  
2. The call to `_lp_shorten_vcs_path` should be in a `if [[ -n "$LP_VCS" ]]` block